Training Plan/BeginningTraining.md

```diff
@@ -0,0 +1,133 @@
+  - FRC Update (Driver Station)
+  - [GitKraken](https://www.gitkraken.com/)
+  - [Git](https://git-scm.com/)
+  - [Radio programming tool](https://docs.wpilib.org/en/latest/docs/software/getting-started/radio-programming.html) (Advanced Students)
+ - [ ] Demo of basic driving robot code
+  - Start with OI and subsystem created, including default command defined
+  - Talk through what is required as input to drive a robot
+  - Let students drive the robot
+
+## Second Meeting:
+ - [ ] Comments
+  - Explain comments
+  - Add single line comment inside drive command
+  - Add multi-line function comment for drive command
+ - [ ] Variables
+  - int
+    - Whole numbers, integars are passed to create motor controllers
+  - float/double
+    - Fractional numbers
+    - Store joystick inputs in variables
+  - boolean
+    - Stores true/false, button inputs
+  - string
+    - Stores text, We use these to name commands
+  - Arrays
+    - Store multiple values of the same type
+    - We'll explore their usage later
+ - [ ] Conditionals
+  - `if`
+    - Implement deadband in drive command
+    - Reduce turning speed at higher forward speed
+
+## Third Meeting:
+ - [ ] Create console programs / Review
+  - Hello world!
+    - Explore printing other things
+  - Basic input
+    - Ask for users name, greet them
+    - Ask users age, print "You can drive" / "You can't drive" as appropriate
+    - Write squaring program
+  - Write a 4 function calculator (Advanced)
+ - [ ] Arrays
+  - Review syntax
+ - [ ] Loops
+  - `for` loop syntax (comparison, iterator syntax)
+    - Fill array by asking for numbers in for loop, print numbers in a separate loop
+    - Print average of numbers gathered above
+  - `while` loop syntax
+    - Simple guessing game, ask user for number until it matches
+    - Explain infinite loop as basis of robot control,
+ - [ ] Averaging Robot exercise
+  - Add averaging to driver input
+  - This is a Low Pass Filter
+  - Explore how increasing the window size (# of samples) slows response
+
+## Fourth Meeting:
+ - [ ] Functions
+  - Way of encapsulating code that is re-used
+    - 3 Components
+      - Type
+      - Name
+      - Parameters
+  - Explore existing functions in subsystem
+    - Command function (returns a command)
+    - Ask students to identify other functions
+  - Robot Examples
+    - Move deadband code into a function
+    - Create function that finds average of an array (replace averaging code from previous session)
+  - Console Examples
+    - Square function
+    - Min function
+    - Max function
+ - [ ] Cover git if there's time remaining
+
+
+## Fifth Meeting
+ - [ ] Classes
+  - Hold information (Member variables) and functionality (methods) that are related
+  - We've already seen classes
+    - Subsystem is a class
+    - Commands are a class
+  - Cover syntax
+    - `public class Name {}`
+    - Constructor, has the same name as the class
+    - Where do Member variables go?
```

Comment:
Make sure we tell them that they should go up at the top of the class - it's annoying trying to find variables and constants that are mixed in with the methods.